pkg/name/ref_test.go

```diff
@@ -18,6 +18,35 @@ import (
 	"testing"
 )
 
+var (
+	testDefaultRegistry = "registry.upbound.io"
+	testDefaultTag      = "stable"
+	inputDefaultNames   = []string{
+		"crossplane/provider-gcp",
+		"crossplane/provider-gcp:v0.14.0",
+		"ubuntu",
+		"gcr.io/crossplane/provider-gcp:latest",
+	}
+	outputDefaultNames = []string{
+		"registry.upbound.io/crossplane/provider-gcp:stable",
+		"registry.upbound.io/crossplane/provider-gcp:v0.14.0",
+		"registry.upbound.io/ubuntu:stable",
```

Comment:
One thing I hadn't considered is how this interacts with dockerhub's `library/` stuff, see here: https://github.com/google/go-containerregistry/blob/ef0e499de80f001ac57881689e9a40a2497247a6/pkg/name/repository.go#L42

If I try to use this for something like mirroring to mirror.example.com, I'd need some way to make `ubuntu` produce `mirror.example.com/library/ubuntu`.

I don't know that this matters for now, perhaps we can add a `WithDefaultNamespace` option if I actually try to do this.

Reply:
Yeah the `WithDefaultNamespace` would behave a bit different, but I think it could certainly be added. Happy to follow-up with that / also look at mirroring more holistically.

Reply:
Let's punt on this and look at it later if we need to.